client/my-sites/site-settings/controller.js

```diff
@@ -118,31 +115,53 @@ module.exports = {
 	},
 
 	deleteSite( context ) {
-		let site = sites.getSelectedSite();
+		const { dispatch, getState } = context.store;
+		const siteSlug = getSelectedSiteSlug( getState() );
+		const siteId = getSelectedSiteId( getState() );
 
-		if ( sites.initialized ) {
-			if ( ! canDeleteSite( site ) ) {
-				return page( '/settings/general/' + site.slug );
+		const redirectIfCantDelete = () => {
```

Comment:
How about moving that one outside, maybe in a separate controller property, so it doesn't get declared every time (and to keep it DRY)?